beets/library.py

```diff
@@ -1369,22 +1369,26 @@ def set_art(self, path, copy=True):
 
         plugins.send('art_set', album=self)
 
-    def store(self, fields=None):
+    def store(self, fields=None, inherit=True):
         """Update the database with the album information.
 
-        The album's tracks are also updated.
-
         `fields` represents the fields to be stored. If not specified,
         all fields will be.
+
+        The album's tracks are also updated when the `inherit` flag is enabled.
+        This applies to fixed attributes as well as flexible ones. The `id`
+        attribute of the album will never be inherited.
         """
         # Get modified track fields.
         track_updates = {}
         track_deletes = set()
         for key in self._dirty:
-            if key in self.item_keys:
+            if key in self.item_keys and inherit:  # Fixed attr
                 track_updates[key] = self[key]
-            elif key not in self:
+            elif key not in self and inherit:  # Fixed or flex attr
                 track_deletes.add(key)
+            elif key != 'id' and inherit:  # Could be a flex attr or id (fixed)
+                track_updates[key] = self[key]
```

Comment:
I think you alluded to this in the main comment thread, but this could be a little easier to read if we put the entire stanza under a `if inherit:` condition. That would make it clear to the reader that _nothing_ happens here when `inherit=False`.

Reply:
Good point! I changed it to this: https://github.com/beetbox/beets/pull/4823/files#diff-d6c57a8deb6bf406de08bd231833fafb4515c984bf0d854a50a0e6cb3dcd3d1fR1386-R1392